### PR TITLE
Fix mightProducePrintableCharacter numeric keypad support

### LINKED ISSUE #71134: Can't use numeric keyboard for search on macOS
Issue Type: <b>Bug</b>

When I start typing file name in file explorer **iec104** and type **104** on addition keyboard (right from arrows) **VSC** didn't catch that input.

If I type **104** on main keyboard (numbers on top of qwerty keys) **VSC** catch that input and find my file.

VS Code version: Code 1.32.3 (a3db5be9b5c6ba46bb7555ec5d60178ecc2eaae4, 2019-03-14T23:38:49.842Z)
OS version: Linux x64 4.15.0-46-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz (8 x 3617)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>surface_synchronization: enabled_on<br>video_decode: unavailable_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|15.65GB (0.36GB free)|
|Process Argv|--unity-launch|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
doxdocgen|csc|0.4.1
snort|inf|1.1.4
autoconf|mae|0.0.7
cpptools|ms-|0.22.1
bash-beautify|sha|0.1.1
markdown-all-in-one|yzh|2.2.0

</details>

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #71134.
I tested it on ubuntu 18.04. Needs to be tested on windows.

### CODE REVIEW

**Comment by joaomoreno** [comment]
@alexandrudima I need your expertise here to figure out if this is a smart thing to do.

File: src/vs/workbench/services/keybinding/browser/keybindingService.ts [hunk 1]
```diff
@@ -535,6 +535,30 @@ export class WorkbenchKeybindingService extends AbstractKeybindingService {
 			return false;
 		}
 		const code = ScanCodeUtils.toEnum(event.code);
+
+		const NUMPAD_PRINTABLE_SCANCODES = [
```
Comment (joaomoreno) [inline src/vs/workbench/services/keybinding/browser/keybindingService.ts]: This should not be allocated every time the `mightProducePrintableCharacter` function runs.

File: src/vs/workbench/services/keybinding/browser/keybindingService.ts [hunk 2]
```diff
@@ -535,6 +535,30 @@ export class WorkbenchKeybindingService extends AbstractKeybindingService {
...
+			ScanCode.Numpad7,
+			ScanCode.Numpad8,
+			ScanCode.Numpad9,
+			ScanCode.Numpad0,
+			ScanCode.NumpadDecimal
+		];
+		const immutableScanCode = IMMUTABLE_KEY_CODE_TO_CODE[event.keyCode];
+		if (NUMPAD_PRINTABLE_SCANCODES.indexOf(code) >= 0 && code === immutableScanCode) {
+			// code === immutableScanCode when NumLock is enabled
+			return true;
```
Comment (joaomoreno) [inline src/vs/workbench/services/keybinding/browser/keybindingService.ts]: We should rather move the condition to the end of the method and ask the opposite question.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

**Comment by joaomoreno** [comment]
@alexandrudima I need your expertise here to figure out whether this is a smart thing to do.

**Comment by R4PH4** [comment]
Hello, do you have news for this bug to be fixed ?

**Comment by R4PH4** [comment]
Hello, do you have news for this bug to be fixed ?

**Comment by R4PH4** [comment]
Do you have news please ?

**Comment by alexdima** [comment]
Thank you!

**Comment by joaomoreno** [comment]
Thanks @alexdima!

**Pull request merged**